internal/status/templates/traces.html

```diff
@@ -38,46 +38,83 @@
   <header class="navbar">
     <a href="/">{{.Tool}} dashboard</a>
   </header>
-
+  <script type="text/javascript">
+    // The code below largely taken from:
+    //   https://perfetto.dev/docs/visualization/deep-linking-to-perfetto-ui
+    const ORIGIN = 'https://ui.perfetto.dev';
+    
+    async function fetchAndOpen(traceUrl) {
+      const resp = await fetch(traceUrl);
+      const blob = await resp.blob();
+      const arrayBuffer = await blob.arrayBuffer();
+      openTrace(arrayBuffer, traceUrl);
+    }
+    
+    function openTrace(arrayBuffer, traceId, traceUrl) {
+      const win = window.open(ORIGIN);
+      if (!win) {
+        alert('Popups blocked. Please allow popups in order to be able to' +
+              'see traces');
+      }
+      const timer = setInterval(() => win.postMessage('PING', ORIGIN), 50);
```

Comment:
50ms seems a bit aggressive? I wonder if we can reliably expect a response in 50ms. I think a few seconds may be fine? If the server responds quickly, we should be able to continue immediately?

Reply:
Unfortunately, a larger delay causes the trace display to be delayed by the passed-in duration (e.g., 5 seconds).

I think what's happening is that `postMessage()` fails the first time while the window is opening, and we end up waiting for the full interval before retrying.

There's no harm in pinging the new window at a higher rate: we stop the pinging and ignore replies when we receive the first response.

I've left it at 50ms for now.